relay-replays/src/recording/mod.rs

```diff
@@ -1,487 +1,134 @@
-}
-
-#[derive(Debug, Serialize, Deserialize)]
-struct BreadcrumbPayload {
-    #[serde(rename = "type")]
-    ty: String,
-    timestamp: f64,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    category: Option<String>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    level: Option<String>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    message: Option<String>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    data: Option<Value>,
-}
-
-#[derive(Debug, Serialize, Deserialize)]
-struct PerformanceSpan {
-    tag: String,
-    payload: PerformanceSpanPayload,
-}
-
-#[derive(Debug, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-struct PerformanceSpanPayload {
-    op: String,
-    description: String,
-    start_timestamp: f64,
-    end_timestamp: f64,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    data: Option<Value>,
-}
-
-/// Node Type Parser
-///
-/// Nodes have an internally tagged variant on their "type" field. The type must be one of six
-/// values.  There are no default types for this variation. Because the "type" field's values are
-/// integers we must define custom deserailization behavior.
-///
-/// -> DOCUMENT = 0
-/// -> DOCUMENTTYPE = 1
-/// -> ELEMENT = 2
-/// -> TEXT = 3
-/// -> CDATA = 4
-/// -> COMMENT = 5
-
-#[derive(Debug, Deserialize, Serialize)]
-#[serde(rename_all = "camelCase")]
-struct Node {
-    #[serde(skip_serializing_if = "Option::is_none")]
-    root_id: Option<i32>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    is_shadow_host: Option<bool>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    is_shadow: Option<bool>,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    compat_mode: Option<String>,
-    #[serde(flatten)]
-    variant: NodeVariant,
+impl From<serde_json::Error> for ParseRecordingError {
+    fn from(err: serde_json::Error) -> Self {
+        ParseRecordingError::Json(err)
+    }
 }
 
-#[derive(Debug)]
-
-enum NodeVariant {
-    T0(Box<DocumentNode>),
-    T1(Box<DocumentTypeNode>),
-    T2(Box<ElementNode>),
-    T3(Box<TextNode>), // text
-    T4(Box<TextNode>), // cdata
-    T5(Box<TextNode>), // comment
+static REPLAY_PII_CONFIG: Lazy<PiiConfig> = Lazy::new(|| {
+    let mut pii_config = PiiConfig::default();
+    pii_config.applications = [(SelectorSpec::And(vec![]), vec!["@common".to_string()])].into();
+    pii_config
+});
+
+static REPLAY_PII_STATE: Lazy<ProcessingState> = Lazy::new(|| {
+    ProcessingState::root().enter_static(
+        "",
+        Some(Cow::Owned(FieldAttrs::new().pii(Pii::True))),
+        Some(ValueType::String),
+    )
+});
+
+fn scrub_string(value: &mut String) {
+    let mut processor = PiiProcessor::new(REPLAY_PII_CONFIG.compiled());
+    match processor.process_string(value, &mut Meta::default(), &REPLAY_PII_STATE) {
+        Err(ProcessingAction::DeleteValueHard) => *value = String::new(),
+        Err(ProcessingAction::DeleteValueSoft) => *value = String::new(),
```

Comment:
Processing "errors" are now handled in-line by removing the value. This is the same behavior as `Annotated` would have.